kiwi/AssocVector.h

```diff
@@ -97,14 +97,10 @@ namespace Loki
 
         typedef C key_compare;
         typedef A allocator_type;
-        typedef typename A::reference reference;
```

Comment:
Out of curiosity why remove those ?

Reply:
These were removed (or moved to` allocator_traits`) in C++20, and already deprecated in 17, see https://en.cppreference.com/w/cpp/memory/allocator